### PR TITLE
PTRENG-4591 add import test to user resources.

### PR DESCRIPTION
**Pull request by danielmkn**
Add import state tests to user resources. 
Move state verification to external go mod `terraform-provider-shared`

### CODE REVIEW

**Review by Turhsus — DISMISSED** [review]
LGTM

File: pkg/artifactory/resource/user/resource_artifactory_user_test.go [hunk 1]
```diff
@@ -2,10 +2,12 @@ package user_test
 
 import (
 	"fmt"
-	"github.com/jfrog/terraform-provider-shared/util"
 	"net/http"
 	"testing"
 
+	"github.com/jfrog/terraform-provider-shared/util"
+	"github.com/jfrog/terraform-provider-shared/validator"
```
Comment (alexhung) [inline pkg/artifactory/resource/user/resource_artifactory_user_test.go:9]: Need sorting

**Review by alexhung — CHANGES_REQUESTED** [review]
One tiny change needed

File: pkg/artifactory/resource/user/resource_artifactory_user_test.go (hunk 1 above)
Comment (danielmkn) [reply pkg/artifactory/resource/user/resource_artifactory_user_test.go:9]: Oh, missed that one, fixed it now.

**Review by Turhsus — APPROVED** [review]
LGTM

**Review by alexhung — APPROVED** [review]
:shipit:

**Pull request merged**